src/core/identifiers/identifiers.test.ts

```diff
@@ -1,20 +1,56 @@
-import { buildDSNPAnnouncementURI, isDSNPAnnouncementURI, parseDSNPAnnouncementURI } from "./identifiers";
+import {
+  buildDSNPAnnouncementURI,
+  convertBigNumberToDSNPUserId,
+  convertBigNumberToDSNPUserURI,
+  convertDSNPUserIdOrURIToBigNumber,
+  convertDSNPUserURIToDSNPUserId,
+  isDSNPAnnouncementURI,
+  isDSNPUserURI,
+  parseDSNPAnnouncementURI,
+} from "./identifiers";
+import { BigNumber } from "ethers";
 
 describe("identifiers", () => {
+  describe("isDSNPUserURI", () => {
+    const validDSNPUserURIs = [
+      "dsnp://0xa123456789abcdef", // Lowercase
+    ];
+
+    const invalidDSNPUserURIs = [
+      "dsnp://0xa123456789ABCDEF", // Uppercase
+      "dsnp://0x0000456789abcdef", // Leading Zeros
+      "dsnp://0123456789abcdef", // No 0x user
+      "dsnp://0xbadwolf", // Bad user URI
+      "dsnp://0xa123456789abcdefa", // URI too long
+    ];
+
+    for (const id of validDSNPUserURIs) {
+      it(`returns true for "${id}"`, () => {
+        expect(isDSNPUserURI(id)).toEqual(true);
+      });
+    }
+
+    for (const id of invalidDSNPUserURIs) {
+      it(`returns false for "${id}"`, () => {
+        expect(isDSNPUserURI(id)).toEqual(false);
+      });
+    }
+  });
+
   describe("isDSNPAnnouncementURI", () => {
     const validDSNPAnnouncementURIs = [
-      "dsnp://0x0123456789ABCDEF/0x0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF", // Uppercase
```

Comment:
why did we drop the uppercase test?

Reply:
The spec says these are invalid. It always specified lowercase and I think we just missed it first time around.